From: Soenke Schwardt-Krummrich <schwardt@univention.de>
Subject: [PATCH] t/kronolith/HK/UV/dateInputFieldOrder

Currently in kronolith the order of date input fields is hardcoded to
year-month-day. Some parts of the world usually use a different order.
The attached patch adds a user preference to specify input field order.

REFERENCE_KOLAB: https://issues.kolab.org/issue3844
REFERENCE_HORDE:

STATUS_KOLAB_2.2:             UNTESTED
STATUS_KOLAB_2.3:             NONE
STATUS_HORDE_KRONOLITH_2.3.3: SUBMIT
STATUS_HORDE_KRONOLITH_3.0:   SUBMIT

Signed-off-by: Gunnar Wrobel <p@rdus.de>

---
 horde-webmail/config/prefs.php                     |   17 ++++++++++-
 horde-webmail/config/prefs.php.dist                |   17 ++++++++++-
 .../kronolith/templates/edit/edit_timespan.inc     |   30 +++++++++++++------
 3 files changed, 52 insertions(+), 12 deletions(-)

diff --git a/templates/edit/edit_timespan.inc b/templates/edit/edit_timespan.inc
index 9926617..05f5042 100644
--- a/templates/edit/edit_timespan.inc
+++ b/templates/edit/edit_timespan.inc
@@ -12,11 +12,16 @@
  </td>
  <td colspan="2" class="rightAlign"><?php echo _("On") ?>&nbsp;</td>
  <td>
-  <?php echo $event->html('start[year]') ?>
-  -
-  <?php echo $event->html('start[month]') ?>
-  -
-  <?php echo $event->html('start[day]') ?>
+ <?php
+     $dateorder = $GLOBALS['prefs']->getValue('date_input_format');
+     $dateorder = $dateorder != "" ? $dateorder : "year-month-day";
+     $dateorder = explode("-", $dateorder );
+     $datefields = array();
+     foreach ($dateorder as $fld) {
+	   $datefields[] = $event->html('start['.$fld.']');
+	 }
+     echo implode("-", $datefields);
+ ?>
 <?php if ($GLOBALS['browser']->hasFeature('dom')): ?>
   <span id="start_wday"></span>
   <script type="text/javascript">
@@ -62,11 +67,16 @@
   <?php echo Horde::label('end_year', _("On")) ?>&nbsp;
  </td>
  <td>
-  <?php echo $event->html('end[year]') ?>
-  -
-  <?php echo $event->html('end[month]') ?>
-  -
-  <?php echo $event->html('end[day]') ?>
+ <?php
+     $dateorder = $GLOBALS['prefs']->getValue('date_input_format');
+     $dateorder = $dateorder != "" ? $dateorder : "year-month-day";
+     $dateorder = explode("-", $dateorder );
+     $datefields = array();
+     foreach ($dateorder as $fld) {
+	   $datefields[] = $event->html('end['.$fld.']');
+	 }
+     echo implode("-", $datefields);
+ ?>
 <?php if ($GLOBALS['browser']->hasFeature('dom')): ?>
   <span id="end_wday"></span>
   <script type="text/javascript">
-- 
tg: (8d3a749..) t/kronolith/HK/UV/dateInputFieldOrder (depends on: t/kronolith/HK/GW/ExportEventList)
-- 
TOPGIT patch commit log
=======================

commit 09e3d2df656b820ae7fbaacce1060958e4b889b9
Author: Gunnar Wrobel <p@rdus.de>
Date:   Thu Nov 5 09:26:25 2009 +0100

    Imported patch from kolab/issue3844 (New user preference to specify order of date input fields)
